packages/state/recoil/selectors/contracts/CommonNft.ts

```diff
@@ -45,6 +47,38 @@ export const queryClient = selectorFamily<
   dangerouslyAllowMutability: true,
 })
 
+export const allNftUsdValueSelector = selectorFamily<
+  number,
+  WithChainId<{ address: string }>
+>({
+  key: 'commonNftAllNftUsdValue',
+  get:
+    ({ chainId, address }) =>
+    ({ get }) => {
+      const accounts = get(accountsSelector({ chainId, address }))
+      let sum = 0
+
+      accounts.forEach((x) => {
+        switch (x.chainId) {
+          case ChainId.StargazeMainnet:
+          case ChainId.StargazeTestnet:
+            sum += get(
+              stargazeWalletUsdValueSelector({
+                chainId: x.chainId,
+                address: x.address,
+              })
+            )
+            break
+
+          default:
+            break
+        }
+      })
+
+      return sum
```

Comment:
a couple things:

first, this `CommonNft` file is for standardizing smart contract queries between cw721 and sg721 contracts, so this selector should probably be in `packages/state/recoil/selectors/nft.ts`. after moving this, you might want to change the `key` too.

also, recoil has this awesome [`waitForAll`](https://recoiljs.org/docs/api-reference/utils/waitForAll/) utility that could parallelize this request:

```suggestion
      const sum = get(
        waitForAll(
          accounts
            .filter(
              (x) =>
                x.chainId === ChainId.StargazeMainnet ||
                x.chainId === ChainId.StargazeTestnet
            )
            .map((x) =>
              stargazeWalletUsdValueSelector({
                chainId: x.chainId,
                address: x.address,
              })
            )
        )
      ).reduce((acc, x) => acc + x, 0)

      return sum
```